RAT/project.py

```diff
@@ -145,17 +166,33 @@ def model_post_init(self, __context: Any) -> None:
                                                                 fit=True, prior_type=RAT.models.Priors.Uniform, mu=0,
                                                                 sigma=np.inf))
 
+        self._all_names = self.get_all_names()
+
         # Wrap ClassList routines - when any of these routines are called, the wrapper will force revalidation of the
         # model, handle errors and reset previous values if necessary.
-        class_lists = ['parameters', 'bulk_in', 'bulk_out', 'qz_shifts', 'scalefactors', 'background_parameters',
-                       'backgrounds', 'resolution_parameters', 'resolutions', 'custom_files', 'data', 'layers',
-                       'contrasts']
         methods_to_wrap = ['_setitem', '_delitem', '_iadd', 'append', 'insert', 'pop', 'remove', 'clear', 'extend']
         for class_list in class_lists:
             attribute = getattr(self, class_list)
             for methodName in methods_to_wrap:
                 setattr(attribute, methodName, self._classlist_wrapper(attribute, getattr(attribute, methodName)))
 
+    @model_validator(mode='after')
+    def update_renamed_models(self) -> 'Project':
```

Comment:
This function scares me with all the nested for loops. Is it possible to implement using signals/slot (observer pattern) instead?